xmtp_api_d14n/src/middleware/auth.rs

```diff
@@ -0,0 +1,348 @@
+use std::sync::Arc;
+
+use arc_swap::ArcSwap;
+use prost::bytes::Bytes;
+use tokio::sync::OnceCell;
+use xmtp_common::{BoxDynError, MaybeSend, MaybeSync, time::now_secs};
+use xmtp_proto::api::{ApiClientError, Client, IsConnectedCheck};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Credential {
+    name: http::header::HeaderName,
```

Comment:
We could probably set a sane default of `Authorization` here so people don't have to think about it unless they want to do something more clever.